### PR TITLE
Cloud Docker Repo Guides

### PR DESCRIPTION
**Pull request by dchw**
- [x] Amazon
 - [ ] Google
 - [ ] Azure

Others?

### CODE REVIEW

File: docs/guides/pushing-images-to-ECR.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Pushing images to ECR
```
Comment (vladaionescu) [inline docs/guides/pushing-images-to-ECR.md:1]:
```suggestion
# Pushing images to AWS ECR
```

Comment (vladaionescu) [inline docs/guides/pushing-images-to-ECR.md:1]: BTW, I would include both pushing and pulling in this guide. Many people might also use `FROM ...` and the setup is pretty much the same.

File: docs/guides/pushing-images-to-ECR.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+## Introduction
+
+Amazons' Elastic Container Registry (ECR) is a hosted docker repository that requires extra configuration for day-to-day use. This configuration is not typical of other repositories, and there are some considerations to account for when using it with Earthly. This guide will walk you through creating an Earthfile, building an image, and pushing it to ECR.
```
Comment (vladaionescu) [inline docs/guides/pushing-images-to-ECR.md:5]:
```suggestion
The AWS Elastic Container Registry (ECR) is a hosted docker repository that requires extra configuration for day-to-day use. This configuration is not typical of other repositories, and there are some considerations to account for when using it with Earthly. This guide will walk you through creating an Earthfile, building an image, and pushing it to ECR.
```

File: docs/guides/pushing-images-to-ECR.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+build:
+    RUN echo "Hello from Earthly!" > motd
+    ENTRYPOINT cat motd
+    SAVE IMAGE --push <aws_account_id>.dkr.ecr.<region>.amazonaws.com/hello-earthly:with-love
+```
+
+## Install and Configure the ECR Credential Helper
+
+ECR does not issue permanent credentials. Instead, it relies on your AWS credentials to issue docker credentials. You can follow instructions [here](https://docs.aws.amazon.com/cli/latest/reference/ecr/get-login.html) to log in with generated credentials, but the process will need to be repeated every 12 hours. In practice, this often means lots of glue code in your CI pipeline to keep credentials up to date.
```
Comment (vladaionescu) [inline docs/guides/pushing-images-to-ECR.md:22]: (Apparently this kind of linking is better for SEO)

```suggestion
ECR does not issue permanent credentials. Instead, it relies on your AWS credentials to issue docker credentials. You can follow [instructions to log in with generated credentials](https://docs.aws.amazon.com/cli/latest/reference/ecr/get-login.html), but the process will need to be repeated every 12 hours. In practice, this often means lots of glue code in your CI pipeline to keep credentials up to date.
```

File: docs/guides/pushing-images-to-ECR.md (hunk 1 above)
Comment (vladaionescu) [inline docs/guides/pushing-images-to-ECR.md:1]: We should link this page from `guides/auth.md` - it'll help people find this.

Also, shall we put this in the TOC? Not sure how to best categorize guides that are not related to the core functionality (but nevertheless very important).

**Review by vladaionescu — COMMENTED** [review]
This looks great! I would merge this independently from other cloud platform guides, given that it's pretty much ready.

File: docs/guides/pushing-images-to-ECR.md (hunk 1 above)
Comment (dchw) [reply docs/guides/pushing-images-to-ECR.md:1]: Yep was planning on it once I finished the other two. Just wanted to get the raw content up for eyes on it first.

**Review by agbell — APPROVED** [review]

**Review by vladaionescu — COMMENTED** [review]
Looks great!

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**